engine/core/src/main/java/org/datacleaner/job/builder/AnalysisJobBuilder.java

```diff
@@ -774,9 +785,11 @@ public AnalysisJobMetadata createMetadata() {
      * 
      * @return
      * @throws IllegalStateException
-     *             if the job is invalidly configured.
+     *             if the job is invalidly configured. See
+     *             {@link #isConfigured(boolean)} for detailed exception
+     *             descriptions.
      */
-    public AnalysisJob toAnalysisJob() throws IllegalStateException {
+    public AnalysisJob toAnalysisJob() throws RuntimeException {
```

Comment:
Hmmmm, as far as I remember best practice is more or less only to have RuntimeExceptions in ```@throws```. At least it seems to be the way their own code is used, and I think best practice would be not to having people guessing if it is a checked exception or not (of course that matters a little less now, with the IDE checking it live).